src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -19,28 +21,73 @@ export function ThemeProvider({
   children,
 }: ThemeProviderProps) {
   const {unstableGlobalTheming = false} = useFeatures();
+
+  const parentContext = useContext(ThemeContext);
+  const isParentThemeProvider = parentContext === undefined;
+  const parentMode = parentContext && parentContext.mode && parentContext.mode;
+  const parentColors =
+    parentContext &&
+    parentContext.UNSTABLE_colors &&
+    parentContext.UNSTABLE_colors;
+
+  const {UNSTABLE_colors, mode, ...rest} = themeConfig;
+
+  const childShouldInheritParentColors =
+    !isParentThemeProvider && mode !== undefined && mode !== parentMode;
+
+  const defaultColors = {
```

Comment:
Can we make `UNSTABLE_Color` be this object instead of an enum? That way you can spread the object instead of reconstructing a whole one if you ever want to create a theme that only manipulates a single role. That makes it a lot more composable:

```
const myPalette = {...UNSTABLE_Color, primary: '#0000ff';}
```

is easier than 

```
const myPalette = {
    surface: UNSTABLE_Color.Surface,
    onSurface: UNSTABLE_Color.OnSurface,
    interactive: UNSTABLE_Color.Interactive,
    neutral: UNSTABLE_Color.Neutral,
    primary: '#0000ff',
    critical: UNSTABLE_Color.Critical,
    warning: UNSTABLE_Color.Warning,
    highlight: UNSTABLE_Color.Highlight,
    success: UNSTABLE_Color.Success,
    decorative: UNSTABLE_Color.Decorative,
}
```

Reply:
@BPScott `theme` rather than `palette`. The word "theme" (what our prop and component is called 🤦‍♂) was completely escaping me yesterday when we were chatting.